controller/src/main/java/org/jboss/as/controller/operations/global/GlobalOperationHandlers.java

```diff
@@ -772,55 +815,47 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
-            execute(aliasAddr == null ? address : aliasAddr, PathAddress.EMPTY_ADDRESS, context);
-        }
+        protected void executeMultiTargetChildren(PathAddress base, PathElement currentElement, PathAddress newRemaining, OperationContext context, ImmutableManagementResourceRegistration registration, boolean ignoreMissing) {
+            final String childType = currentElement.getKey().equals("*") ? null : currentElement.getKey();
+            if (registration.isRemote()) {// || registration.isRuntimeOnly()) {
+                // At least for proxies it should use the proxy operation handler
+                throw new IllegalStateException();
+            }
 
-        void execute(final PathAddress address, PathAddress base, final OperationContext context) {
-            final PathAddress current = address.subAddress(base.size());
-            final Iterator<PathElement> iterator = current.iterator();
-            if (iterator.hasNext()) {
-                final PathElement element = iterator.next();
-                if (element.isMultiTarget()) {
-                    final Set<PathElement> children = context.getResourceRegistration().getChildAddresses(base);
-                    if (children == null || children.isEmpty()) {
-                        return;
-                    }
-                    final String childType = element.getKey().equals("*") ? null : element.getKey();
-                    for (final PathElement path : children) {
-                        if (childType != null && !childType.equals(path.getKey())) {
-                            continue;
-                        }
-                        execute(address, base.append(path), context);
-                    }
-                } else {
-                    execute(address, base.append(element), context);
-                }
-            } else {
-                //final String operationName = operation.require(OP).asString();
-                final ModelNode newOp = operation.clone();
-                newOp.get(OP_ADDR).set(base.toModelNode());
+            final Set<PathElement> children = context.getResourceRegistration().getChildAddresses(base);
+            if (children == null || children.isEmpty()) {
+                return;
+            }
 
-                final ModelNode result = this.result.add();
-                result.get(OP_ADDR).set(base.toModelNode());
-                context.addStep(result, newOp, handler, OperationContext.Stage.MODEL, true);
+            boolean isServerWildcardWithRemaining = currentElement.getKey().equals(RUNNING_SERVER) && currentElement.isWildcard() && newRemaining.size() > 0;
+            for (final PathElement path : children) {
+                if (childType != null && !childType.equals(path.getKey())) {
+                    continue;
+                }
+                if (isServerWildcardWithRemaining && path.getKey().equals(RUNNING_SERVER) && path.isWildcard() && newRemaining.size() > 0) {
```

Comment:
This confuses me. L837 is doing a similar check on "currentElement" as is being done here on "path". And the newRemaining.size() > 0 check is redundant.